bee-network/bee-autopeering/src/lib.rs

```diff
@@ -0,0 +1,145 @@
+// Copyright 2021 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+//! Allows peers in the same IOTA network to automatically discover each other.
+//!
+//! In order to integrate the Autopeering functionality in your node implementation you need to provide its `init`
+//! function with the following data:
+//! * an `AutopeeringConfig`;
+//! * a protocol version (`u32`);
+//! * a network name, e.g. "chrysalis-mainnet";
+//! * a `Local` entity (either randomly created or from an `Ed25519` keypair), that additionally announces one or more
+//!   services;
+//! * a shutdown signal (`Future`);
+//! * a peer store, e.g. the `InMemoryPeerStore` (non-persistent) or the `SledPeerStore` (persistent), or a custom peer
+//!   store implementing the `PeerStore` trait;
+//!
+//! ## Example
```

Comment:
Is this the exact same example than in the example folder ? Can't we just refer to it ? Feels weird to have it duplicated and have a fully fledged example here in the crate root.

Reply:
They are pretty similar atm, but they don't need to be. Judging from my own behavior when investigating a crate I first have a look at `docs.rs`, and I am always thankful if the crate author put some example right into `lib.rs`, so I can get a first impression of the API. If I like the crate I take the next step and look at the examples and maybe run a few them to see how the code actually behaves. The `lib.rs` documentation doesn't have to be runnable, and I probably will strip the example given there down even more. 
tl;dr
I want both examples to be able to evolve separately and spare the user the redirection.